cmd/cloud-controller-manager/app/config/config.go

```diff
@@ -17,25 +17,39 @@ limitations under the License.
 package app
 
 import (
-	"time"
-
+	apiserver "k8s.io/apiserver/pkg/server"
+	clientset "k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/record"
 	genericcontrollermanager "k8s.io/kubernetes/cmd/controller-manager/app"
+	"k8s.io/kubernetes/pkg/apis/componentconfig"
 )
 
-// ExtraConfig are part of Config, also can place your custom config here.
-type ExtraConfig struct {
-	NodeStatusUpdateFrequency time.Duration
-}
-
 // Config is the main context object for the cloud controller manager.
 type Config struct {
```

Comment:
The word `Config` is heavily overloaded, and doesn't really capture that this type holds runtime state. Would `Runtime` or `Server` be more appropriate?

Reply:
This PR is mainly a giant cleanup of the structure. We will discuss the future patterns, with your kubelet experience in mind in follow-ups. So for now this resembles the terms we use in apiservers.

Reply:
SGTM.